### DISCUSSION TITLE
Suggestion: Make an objective target always an array

### DISCUSSION THREAD

**Discussion by kokarn** [TarkovTracker/tarkovdata#114] [Schema Changes]
It kinda doesn't make sense to have it as a string sometimes and as an array sometimes

Current examples
```javascript
[{
    "type": "collect",
    "target": "54491c4f4bdc2db1078b4568",
    "number": 2,
    "location": "Any",
    "id": 1
},
            
{
"type": "key",
    "target": [
        "5a13ef0686f7746e5a411744",
        "5a0ee34586f774023b6ee092"
    ],
    "number": 1,
    "location": "Shoreline",
    "id": 142
}]
```

Suggested change
```javascript
[{
    "type": "collect",
    "target": [
        "54491c4f4bdc2db1078b4568"
    ],
    "number": 2,
    "location": "Any",
    "id": 1
},
            
{
"type": "key",
    "target": [
        "5a13ef0686f7746e5a411744",
        "5a0ee34586f774023b6ee092"
    ],
    "number": 1,
    "location": "Shoreline",
    "id": 142
}]
```

**Comment by thaddeus** [comment]
Agreed 👍

**Comment by thaddeus** [comment]
I've been thinking about this some more. Keys right now are the only thing that the arrays apply to for targets. Perhaps instead change the target to something like the map, room, or even the coordinates the key(s) are needed when maps coordinates are added?

Instead the key lists themselves would be something like the `with` array. I do want to rework the `with` arrays to be structured more like `with: [{type: "wear", target: "item_id_goes_here"}, {type: "hit", target: "legs"}, {type: "key", target: "item_id_goes_here"}]` and you could have multiple keys for the few situations where you can use one of multiple resort keys (I'm guessing streets will have at least a couple similarities here)

One other potential related solution would be to change the appropriate objectives to utilize the `with type key` style data and remove key objectives entirely. But that probably would make it more complex to access what keys you potentially need for quests.
